src/pipdeptree/_models/dag.py

```diff
@@ -110,16 +109,11 @@ def filter_nodes(self, include: list[str] | None, exclude: set[str] | None) -> P
         if include is None and exclude is None:
             return self
 
-        # Note: In following comparisons, we use lower cased values so
-        # that user may specify `key` or `project_name`. As per the
-        # documentation, `key` is simply
-        # `project_name.lower()`. Refer:
-        # https://setuptools.readthedocs.io/en/latest/pkg_resources.html#distribution-objects
         include_with_casing_preserved: list[str] = []
         if include:
             include_with_casing_preserved = include
-            include = [s.lower() for s in include]
-        exclude = {s.lower() for s in exclude} if exclude else set()
+            include = [pep503_normalize(i) for i in include]
+        exclude = {pep503_normalize(s) for s in exclude} if exclude else set()
```

Comment:
Since we changed this normalize user input now, we should add a test for this to make sure that it does so in the future.